modules/order/src/Plugin/Field/FieldType/AdjustmentItem.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\commerce_order\Plugin\Field\FieldType;
+
+use Drupal\commerce_order\Adjustment;
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\TypedData\MapDataDefinition;
+
+/**
+ * Plugin implementation of the 'commerce_price' field type.
+ *
+ * @dearBojan: instead of just serializing a Adjustment object, put its
+ *              definition items as the schema, and it can return. This way
+ *              we could actually expose the data to Views and others.
+ *
+ * @FieldType(
+ *   id = "commerce_order_adjustment",
```

Comment:
Shouldn't the plugins be just commerce_adjustment, not commerce_order_adjustment? We are going to use them for line items too, no?